PR: https://github.com/odoo/odoo/pull/127032

From: 01eb3b008af788f80e2000f91d46a386d6ea5465
From: kusi-odoo
Date: 2024-01-03 15:07:50

Breaking data model changes score: 6, change matches:
-    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills")
+    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills",
-    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills")
+    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills",

Total Changes: 7

[IMP] hr_skill: add archive/Unarchive skill type functionality in employee app

In this commit we have added archive/unarchive functionality with ribbon in the
skill type.so user can easily archive/Unarchive the skill type.

task-3385297

closes odoo/odoo#127032

Related: odoo/enterprise#45743
Signed-off-by: Sofie Gvaladze (sgv) <sgv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_skills/models/hr_employee.py
+++ b/addons/hr_skills/models/hr_employee.py
@@ -8,7 +8,8 @@ class Employee(models.Model):
     _inherit = 'hr.employee'
 
     resume_line_ids = fields.One2many('hr.resume.line', 'employee_id', string="Resume lines")
-    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills")
+    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills",
+        domain=[('skill_type_id.active', '=', True)])
     skill_ids = fields.Many2many('hr.skill', compute='_compute_skill_ids', store=True)
 
     @api.depends('employee_skill_ids.skill_id')

--- a/addons/hr_skills/models/hr_employee_public.py
+++ b/addons/hr_skills/models/hr_employee_public.py
@@ -8,4 +8,5 @@ class EmployeePublic(models.Model):
     _inherit = 'hr.employee.public'
 
     resume_line_ids = fields.One2many('hr.resume.line', 'employee_id', string="Resume lines")
-    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills")
+    employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills",
+        domain=[('skill_type_id.active', '=', True)])

--- a/addons/hr_skills/models/hr_skill_type.py
+++ b/addons/hr_skills/models/hr_skill_type.py
@@ -9,6 +9,7 @@ class SkillType(models.Model):
     _description = "Skill Type"
     _order = "name"
 
+    active = fields.Boolean('Active', default=True)
     name = fields.Char(required=True)
     skill_ids = fields.One2many('hr.skill', 'skill_type_id', string="Skills")
     skill_level_ids = fields.One2many('hr.skill.level', 'skill_type_id', string="Levels")
